icu4c/source/i18n/number_formatimpl.cpp

```diff
@@ -524,15 +528,30 @@ int32_t NumberFormatterImpl::writeNumber(const MicroProps& micros, DecimalQuanti
 
         // Add the decimal point
         if (quantity.getLowerDisplayMagnitude() < 0 || micros.decimal == UNUM_DECIMAL_SEPARATOR_ALWAYS) {
-            length += string.insert(
+            if (micros.currencyAsDecimal) {
+                length += string.insert(
                     length + index,
-                    micros.useCurrency ? micros.symbols->getSymbol(
-                            DecimalFormatSymbols::ENumberFormatSymbol::kMonetarySeparatorSymbol) : micros
-                            .symbols
-                            ->getSymbol(
-                                    DecimalFormatSymbols::ENumberFormatSymbol::kDecimalSeparatorSymbol),
+                    // Note: This unconditionally substitutes the standard short symbol.
+                    // TODO: Should we support narrow or other variants?
```

Comment:
Yes, we should support at least narrow and standard short. And given that, I think it is cleanest and most understandable to support all.

Reply:
Okay. I wrote the code this way since the short symbol was easy to access at this point in the code. But I'll go ahead and rewire things so that I can get access to other symbol widths from here.

Reply:
Did you want to submit this first for now, and then follow this up under the same ticket to support all the symbols?

Reply:
Yes, given the imminent CLDR deadline, the expedient thing would be to check this in, but leave the ticket open, and follow up to finish the ticket before ICU 70.  (This is a very busy week with my intern starting...)

Reply:
Okay, I'm finally addressing this.  So, an issue right now is that we're parsing only a single currency placeholder ('¤') as the decimal separator.  But how do we know whether to use the narrow or formal symbol?

Eventually, once we have this pattern in CLDR, we can do this in NumberFormatter.  However, since patterns are only available in DecimalFormat, we don't currently have a way to select the negative symbol in the ICU API.

But, I will wire it up nonetheless.